doc/src/documentation_style_guide.rst

```diff
@@ -0,0 +1,1019 @@
+===============================
+SymPy Documentation Style Guide
+===============================
+
+**A Writing Resource for Documentation and Docstrings**
+
+.. contents:: Table Of Contents
```

Comment:
I think that table of contents are already added in the left panel.
Would there be a reason to add another TOC below the title?

Reply:
There's also some weird CSS issue with the manual table of contents, where it puts a blank space under bullets that have subbullets. There's a `margin-block-end: 16px`. I'm not sure where it comes from. If we remove this it's not a big deal but it could be worth looking into.

Reply:
Good point @sylee957. The table of contents was for the wiki version of the guide and I forgot that on the SymPy website one is automatically generated in the left panel, so I'll take this out. Thanks for the feedback!